axiom/nr_distributed_trace.c

```diff
@@ -481,14 +481,28 @@ void nr_distributed_trace_set_app_id(nr_distributed_trace_t* dt,
 }
 
 void nr_distributed_trace_set_trace_id(nr_distributed_trace_t* dt,
-                                       const char* trace_id) {
+                                       const char* trace_id,
+                                       bool do_padding) {
   if (NULL == dt) {
     return;
   }
 
   nr_free(dt->trace_id);
   if (trace_id) {
-    dt->trace_id = nr_strdup(trace_id);
+    if (do_padding) {
+      int len = nr_strlen(trace_id);
+      if (len < NR_TRACE_ID_SIZE) {
```

Comment:
If `len` will be equal to `NR_TRACE_ID_SIZE`, `dt->trace_id` will not be set.

Reply:
that's embarrassing. Good find, fixed in [48230b7](https://github.com/newrelic/newrelic-php-agent/pull/929/commits/48230b7d98677955c32f66dbf6d022d5e0f6c685)